app/controllers/status_controller.rb

```diff
@@ -0,0 +1,49 @@
+### This will go away. The logic in here would go into the Inisghts Common Status Controller
+class StatusController < ActionController::API
+  def health
+    @output = {}
+
+    if db_down?
+      @output[:status] = "fail"
+      @output[:database_status] = "down"
+    else
+      @output[:status] = migration_context.needs_migration? ? "warn" : "pass"
+      @output[:database_status] = "up"
+    end
+
+    @output[:pending_migrations] = pending_migrations
+
+    render :json => @output, :status => status
+  end
+
+  private
+
+  def migration_context
+    ActiveRecord::Base.connection.migration_context
+  end
+
+  def pending_migrations
+    migration_context.migrations_status.collect(&:first).count("down")
+  end
+
+  def status
+    case @output[:status]
+    when "fail"
+      :internal_server_error
+    when "warn"
+      :partial_content
+    else
+      :ok
+    end
+  end
+
+  def db_down?
+    ActiveRecord::Base.connection
```

Comment:
Is this the best way to do this? Should we actually query something? Or is there some kind of health check method built into the connection object? The raw `PG::Connection` class has a `.ping` https://www.rubydoc.info/gems/pg/PG/Connection#ping-class_method not sure if that would be good to use here.

Reply:
Oh yeah, I like the idea of that. I have never worked with the PG gem directly, updated the method to do this:
`PG::Connection.ping(ENV['DATABASE_URL']) != PG::Connection::PQPING_OK`